lib/whedon/compilers.rb

```diff
@@ -0,0 +1,242 @@
+# This module has methods to compile PDFs and Crossref XML depending upon
+# the content type of the paper (Markdown or LaTeX)
+module Compilers
+
+  # Generate the paper PDF
+  # Optionally pass in a custom branch name as first param
+  def generate_pdf(custom_branch=nil,paper_issue=nil, paper_volume=nil, paper_year=nil)
+    if paper.latex_source?
+      pdf_from_latex(custom_branch, paper_issue, paper_volume, paper_year)
+    elsif paper.markdown_source?
+      pdf_from_markdown(custom_branch, paper_issue, paper_volume, paper_year)
+    end
+  end
+
+  def generate_crossref(paper_issue=nil, paper_volume=nil, paper_year=nil, paper_month=nil, paper_day=nil)
+    if paper.latex_source?
+      crossref_from_latex(paper_issue=nil, paper_volume=nil, paper_year=nil, paper_month=nil, paper_day=nil)
+    elsif paper.markdown_source?
+      crossref_from_markdown(paper_issue=nil, paper_volume=nil, paper_year=nil, paper_month=nil, paper_day=nil)
+    end
+  end
+
+  def pdf_from_latex(custom_branch=nil,paper_issue=nil, paper_volume=nil, paper_year=nil)
```

Comment:
Ultimately might want to clean up this method - this currently is what the JCON folks need to compile their papers.